### PR TITLE
Use mysql version from server rather than mysql --version

### PR DESCRIPTION
**Pull request by grooverdan**
mysql --version is the version of the client program and not necessarly the server version.

Get the version accurate for the reports generated.

### CODE REVIEW

File: prepare.php [hunk 1]
```diff
@@ -99,6 +98,11 @@
 	echo PHP_EOL;
 }
 EOT;
+$db_version = <<<EODB
+\$mysqli = new mysqli(DB_HOST, DB_USER, DB_PASSWORD, DB_NAME);
```
Comment (pfefferle) [inline prepare.php:102]: this will be called outside of the hosting env, so it might cause issues, because not every DB might be publicly accessible.

Comment (grooverdan) [reply prepare.php:102]: This is is escaped so its forming the code appended to `/wp-tests-config.php` which as far as I know is executed within the hosting environment.

Comment (pfefferle) [reply prepare.php:102]: Ah, I see! but then we change the version from cli version to mysql version. I think this is even better, but does that lead to confusion?

Comment (grooverdan) [reply prepare.php:102]: if its suddenly slightly different people should quickly realize the newer version is actually accurate.

Comment (grooverdan) [reply prepare.php:102]: acceptable?

**Review by pfefferle — APPROVED** [review]
LGTM but not tested!

**Comment by grooverdan** [comment]
Sample output:
```
{
    "php_version": "7.4.33",
    "php_modules": {
        "bcmath": false,
        "curl": "7.4.33",
        "filter": "7.4.33",
        "gd": "7.4.33",
        "libsodium": false,
        "mcrypt": false,
        "mod_xml": false,
        "mysqli": "7.4.33",
        "imagick": false,
        "gmagick": false,
        "pcre": "7.4.33",
        "xml": "7.4.33",
        "xmlreader": "7.4.33",
        "zlib": "7.4.33"
    },
    "system_utils": {
        "curl": "7.74.0 OpenSSL\/1.1.1n 1.2.11",
        "openssl": "1.1.1n  15 Mar 2022"
    },
    "os_name": "Linux",
    "os_version": "6.2.14-300.fc38.x86_64",
    "mysql_version": "10.11.2-MariaDB-1:10.11.2+maria~ubu2204"
}
```

Needs one more change however - putting `file_put_contents( __DIR__ . '/tests/phpunit/build/logs/env.json'` after fetching the the mysql version. I'll append this soon.

**Comment by grooverdan** [comment]
Was appened so the generated file env.json is:
```
cat env.json 
{
    "php_version": "8.2.10",
    "php_modules": {
        "bcmath": false,
        "curl": "8.2.10",
        "dom": "20031129",
        "exif": false,
        "filter": "8.2.10",
        "gd": "8.2.10",
        "gettext": false,
        "gmagick": false,
        "iconv": "8.2.10",
        "imagick": false,
        "intl": false,
        "libsodium": false,
        "libxml": "8.2.10",
        "mbstring": "8.2.10",
        "mcrypt": false,
        "mod_xml": false,
        "mysqli": "8.2.10",
        "pcre": "8.2.10",
        "xml": "8.2.10",
        "xmlreader": "8.2.10",
        "zlib": "8.2.10"
    },
    "system_utils": {
        "curl": "7.88.1 OpenSSL\/3.0.9 1.2.13",
        "openssl": "3.0.9 30 May 2023 (Library: 3.0.9 30 May 2023)"
    },
    "os_name": "Linux",
    "os_version": "6.4.12-200.fc38.x86_64",
    "mysql_version": "11.2.2-MariaDB-1:11.2.2+maria~ubu2204"
}
```

This is a combined version with #177 like https://github.com/WordPress/phpunit-test-runner/compare/master...grooverdan:phpunit-test-runner:all_changes.

**Comment by grooverdan** [comment]
reminder.

**Review by javiercasares — APPROVED** [review]

**Comment by grooverdan** [comment]
reverted and I don't know why :cry: 29af48bf8ed33d1e24b77884e3d635289418d924

**Pull request merged**